### PR TITLE
Maven plugin configuration & goal hover + refactorings

### PR DESCRIPTION
**Pull request by AObuchow**
Signed-off-by: Andrew Obuchowicz aobuchow@redhat.com

### CODE REVIEW

**Comment by AObuchow** [comment]
@mickaelistria if you have a chance to check this out, let me know what you think.

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RemoteRepository.Builder;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.services.extensions.IPositionRequest;
+
+public class Utils {
+
+	public static MarkupContent parameterToHoverMarkup(Parameter parameter) {
```
Comment (AObuchow) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java]: should maybe be renamed to `parameterToMarkup`

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
+package org.eclipse.lsp4xml.extensions.maven;
```
Comment (mickaelistria) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java]: Missing copyright header

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.maven.plugin.descriptor.Parameter;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RemoteRepository.Builder;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4xml.dom.DOMNode;
+import org.eclipse.lsp4xml.services.extensions.IPositionRequest;
+
+public class Utils {
```
Comment (mickaelistria) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java]: We need a clearer name than Utils or this class would become too messy.
I think we should split it into 2 parts `MavenPluginUtils` for the Maven parts to reuse, and `DOMUtils` for the plain DOM parts. In fine, it's likely that we try to replace the `DOMUtils` by some API provided in LemMinX directly if existing, or that we migrate our DOMUtils to LemMinX.

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java (hunk 1 above)
Comment (mickaelistria) [reply org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java]: It doesn't seem to be specifically for Hover, but also for completion, so I suggest we rename this method to `getMarkupDescription(Parameter parameter)`. I don't think we need to keep the "parameterTo" in the method name as it's a single argument method so it's easy enough to get what it does.

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenHoverParticipant.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+package org.eclipse.lsp4xml.extensions.maven;
```
Comment (mickaelistria) [inline org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenHoverParticipant.java]: Copyright header

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java (hunk 3 above)
Comment (AObuchow) [reply org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/Utils.java]: Agreed, maybe we should show `DomUtils` to the LeMiNX upstream sometime in the near future.

File: org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenHoverParticipant.java (hunk 4 above)
Comment (AObuchow) [reply org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenHoverParticipant.java]: good catch

**Pull request merged**